### ISSUE TITLE
Why is there [:,1:,:,:,:] to slice the output ?

### ISSUE THREAD

**Issue by kadircanbecek** [jocpae/clDice#4]

**Comment by jocpae** [comment]
Hi, thank you for your comment.

**Comment by Branikas** [comment]
Hello. Congratulations for your work, we also had a chat during CVPR and it really inspired me with my research as well!
I  think that the uploaded code is not runnable as it is for the pytorch implementation (similarly to the closed issue that was posted ).
I think that the soft_dice_cldice object should inherit from the soft_cldice and not the nn.Module otherwise python raises some errors (I can reproduce them and post a snippet here if you would like).
Also I think it would be extremely helpful for everyone if you could post a minimal working example where you apply your loss on the output of a model, so that everyone can see what the expected shape of the output should be, regarding the one-hot encoding, and how you define your data as foreground and background.
Many thanks for all the help and the great work!

Kind Regards,
Stathis

**Comment by jocpae** [comment]
Hi Stathis, 
please send us the code snippet so we can comment. The code is runnable though and has been tested by many. We agree the inheritance of the classes could be structured in many ways. 

Best regards

**Comment by Branikas** [comment]
Hi jocpae,

Many thanks for the reply. I worked my way around some errors I was getting. My main question (and the reason why I posted the comment on this issue) is that I am not sure about what the shape of the output should be. You mention that you are using one-hot for distinguishing between fore and background and that is why you have 5 indexes.
My output is a 4-D tensor in the standard notation and is essentially a traditional binary mask, so I drew the assumption that the standard binary mask meets the criteria for being the foreground (to my best understanding from your definition of it in the paper) and I remove the slicing.
I am applying this to further train a model that has trained to the maximum extend and provides already satisfying results to see if any improvements are observed, but the model's validation accuracy is 0 from the first epoch.
My question is whether or not applying the model on the "raw" binary predictions is correct or not. In case it is correct I wonder if I have to use some extreme hyper-parameter values (like very small learning rate values for the optimizer).
In any case many thanks for all the help!

Regards,
Stathis
